presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSource.java

```diff
@@ -66,43 +55,15 @@
     public OrcPageSource(
             OrcRecordReader recordReader,
             OrcDataSource orcDataSource,
-            List<HiveColumnHandle> columns,
-            TypeManager typeManager,
+            Map<Integer, Type> includedColumns,
             AggregatedMemoryContext systemMemoryContext,
             FileFormatDataSourceStats stats)
     {
         this.recordReader = requireNonNull(recordReader, "recordReader is null");
         this.orcDataSource = requireNonNull(orcDataSource, "orcDataSource is null");
-
-        int size = requireNonNull(columns, "columns is null").size();
-
-        this.stats = requireNonNull(stats, "stats is null");
-
-        this.constantBlocks = new Block[size];
-        this.hiveColumnIndexes = new int[size];
-
-        ImmutableList.Builder<String> namesBuilder = ImmutableList.builder();
-        ImmutableList.Builder<Type> typesBuilder = ImmutableList.builder();
-        for (int columnIndex = 0; columnIndex < columns.size(); columnIndex++) {
-            HiveColumnHandle column = columns.get(columnIndex);
-            checkState(column.getColumnType() == REGULAR, "column type must be regular");
```

Comment:
Where did this logic go? 
Am I correct this is handled by `HivePageSourceProvider`?

If so, I'd name this commit differently. Something signalling you're simply reusing information already collected (removing unnecessary code). Or, you may include this information as part of commit message body.

Reply:
The useful logic in this block is the checking done by `recordReader.isColumnPresent`. I moved it to `getNextPage`.

Reply:
@findepi I had to do some digging when I reviewed this.  This code is a duplicate of https://github.com/prestosql/presto/blob/f37659ee6ce907e51a75f325ed744c2fcbc459ca/presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSourceFactory.java#L193-L201  I think this is just cruft from refactorings.